src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -49,6 +49,29 @@ class NativeInterface
         private bool Unity2019OrNewer;
         private Thread MainThread;
 
+        private class OnSessionCallback : AndroidJavaProxy
+        {
+
+            private Configuration _config;
+
+            public OnSessionCallback(Configuration config) : base("com.bugsnag.android.OnSessionCallback")
+            {
+                _config = config;
+            }
+            public bool onSession(AndroidJavaObject session)
+            {
+                var wrapper = new NativeSession(session);
+                foreach (var callback in _config.GetOnSessionCallbacks())
+                {
+                    if (!callback.Invoke(wrapper))
```

Comment:
If the callback throws an exception the requirement is to catch it and continue if possible. I think there are also some requirements around logging information - the spec should have more info.

Reply:
Added a handler, could not find any mention of logging in the spec